hikari/interactions/modal_interactions.py

```diff
@@ -0,0 +1,359 @@
+# -*- coding: utf-8 -*-
+# cython: language_level=3
+# Copyright (c) 2020 Nekokatt
+# Copyright (c) 2021-present davfsa
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Models and enums used for Discord's Modals interaction flow."""
+
+from __future__ import annotations
+
+__all__: typing.List[str] = [
+    "ModalResponseTypesT",
+    "ModalInteraction",
+    "InteractionTextInput",
+    "ModalInteraction",
+    "TextInputStyle",
+    "ModalComponentType",
+    "PartialModalComponent",
+    "ModalActionRowComponent",
+]
+
+import typing
+
+import attr
+
+from hikari import channels
+from hikari import guilds
+from hikari import messages
+from hikari import permissions
+from hikari import snowflakes
+from hikari import traits
+from hikari.interactions import base_interactions
+from hikari.internal import attr_extensions
+from hikari.internal import enums
+
+if typing.TYPE_CHECKING:
+    from hikari import users as _users
+    from hikari.api import special_endpoints
+
+ModalResponseTypesT = typing.Literal[
+    base_interactions.ResponseType.MESSAGE_CREATE, 4, base_interactions.ResponseType.DEFERRED_MESSAGE_CREATE, 5
```

Comment:
Hello, I was just testing modals, I believe this type hinting is incorrect. Both DEFERRED_UPDATE_MESSAGE and UPDATE_MESSAGE are valid if the modal was sent by a ComponentInteraction. 
See: [the doc](https://discord.com/developers/docs/interactions/receiving-and-responding#responding-to-an-interaction)